### PR TITLE
Move to GH Actions

### PR DESCRIPTION
**Pull request by unleashed**
This sets up GitHub Actions on this repo.

With this we get rid of Travis, lose Appveyor (we could keep it, but not worth it AFAICT) and also lose uploading master docs to GH pages (which has been broken for a while, but perhaps we can restore it, although I don't think it is particularly useful now that we get docs published in docs.rs).

### CODE REVIEW

**Comment by davidor** [comment]
@unleashed Is this still WIP? What's missing?

**Comment by unleashed** [comment]
No, I think it is ready, what is missing is you approving, then changing the repo settings for required statuses and then bors.

**Comment by davidor** [comment]
You forgot to delete the Travis and Appveyor badges from the README. It'd be good to add one for GH actions.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Note: change to suit your needs.
```
Comment (davidor) [inline .github/CODEOWNERS:1]: :eyes:

Comment (unleashed) [reply .github/CODEOWNERS]: Yes, this comes from my rust-template repo :) - removed.

**Comment by unleashed** [comment]
> You forgot to delete the Travis and Appveyor badges from the README. It'd be good to add one for GH actions.

Done!

**Comment by unleashed** [comment]
This should be ready now. The coverage is pretty broken - this might have something to do with using tarpaulin - but we can look into that later.

**Review by davidor — APPROVED** [review]

**Comment by unleashed** [comment]
Let me perform a couple extra checks with tarpaulin :)

**Comment by unleashed** [comment]
Another bug in GitHub - if my commit is dated before your review, it would appear as if you acked it. :man_dancing: 
image

**Comment by unleashed** [comment]
Ok, so now tarpaulin uses `--all-features` (not that it changed anything in the coverage, but it was a bug), and the format check is extracted out of CI so it is only tested with latest stable (and made it required).

**Comment by unleashed** [comment]
bors retry

**Comment by unleashed** [comment]
ahem... :shrug: 

bors r=@davidor

**Comment by unleashed** [comment]
Ok, missing a check on the merged result... :/

**Comment by unleashed** [comment]
bors retry

**Comment by unleashed** [comment]
This is just minor stuff about the checks - such as not running both pull_request and push checks here which both are going to be the same yet they use two different runs.

**Comment by unleashed** [comment]
Ok, down from 12 to 9 checks in PR. Let's see whether push + master work well.

**Comment by unleashed** [comment]
bors r=@davidor

**Comment by unleashed** [comment]
pim pam!

**Pull request merged**